ui/components/PromptComponent.js

```diff
@@ -60,87 +60,77 @@ const styles = (theme) => ({
   }
 });
 
-class PromptComponent extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      show : false,
-      title : "",
-      subtitle : "",
-      options : []
+function PromptComponent(props) {
+  const [show, setShow] = useState(false);
+  const [title] = useState("");
+  const [subtitle] = useState("");
+  const [options] = useState([]);
+  const [promiseInfo] = useState({ resolve : null, reject : null });
+
+  useEffect(() => {
```

Comment:
Is this `useEffect` really necessary?

Reply:
It's purpose was to listen for changes in the show state, It isn't that necessary

Reply:
Have a look at the change @Rajdip019 , and is anything necessary to update there

Reply:
Looks good I guess.

Reply:
@Rajdip019  could you please perform a final review to ensure everything is in order and, if appropriate, merge the PR?